python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_checks.py

```diff
@@ -40,65 +36,22 @@ def has_asset_checks(
     )
 
 
-# fetch all statuses at once in order to batch the run query
-def get_asset_check_execution_statuses_by_id(
```

Comment:
We no longer have to do this kinda annoying caching stuff, as we can rely on the LoadingContext to keep track of all the things (runs, check records, asset records) we're interested in and therefore not be scared of querying for it multiple times.

We do need to do a bit of a bespoke `prepare` call in fetch_asset_check_executions in order to make sure we fetch all the runs we'll need in a single batch, but other than that everything just happens automatically for us.

Reply:
> We do need to do a bit of a bespoke prepare call in fetch_asset_check_executions in order to make sure we fetch all the runs we'll need in a single batch

should we add some sort of defense or way to know when a prepare should get added? Like a something that warns on a # of blocking_gets for the same type ? A test that spys on the storage method to make sure its only called once?